### PR TITLE
improvement(types): add typescript definitions & refactor to fastify v3

### PR DESCRIPTION
**Pull request by rqbazan**
BREAKING CHANGE: the fastify version was updated from 1.x to 3.x

### CODE REVIEW

**Comment by herrmannplatz** [comment]
Big thanks for you help!

File: package.json [hunk 1]
```diff
@@ -18,16 +19,23 @@
...
   "dependencies": {
-    "fastify-plugin": "1.3.0"
+    "fastify-plugin": "^1.6.1"
   },
   "devDependencies": {
+    "@types/node": "^14.0.14",
     "commitizen": "^3.0.4",
     "cz-conventional-changelog": "^2.1.0",
-    "fastify": "^1.13.1",
+    "fastify": "^3.0.0-rc.5",
```
Comment (herrmannplatz) [inline package.json]: Think we can go for the proper 3.0.0 release now

File: package.json [hunk 2]
```diff
@@ -18,16 +19,23 @@
...
-    "jest": "^23.6.0",
+    "jest": "^26.1.0",
     "semantic-release": "^15.12.0",
-    "standard": "^12.0.1"
+    "standard": "^12.0.1",
+    "tsd": "^0.13.1",
+    "typescript": "^3.9.6"
+  },
+  "peerDependencies": {
+    "fastify": "^3.0.0-rc.5"
```
Comment (herrmannplatz) [inline package.json]: same here

**Comment by rqbazan** [comment]
Hey @herrmannplatz  I've just updated Fastify 🎉

**Review by herrmannplatz — APPROVED** [review]

**Pull request merged**